.github/workflows/health_metrics.yml

```diff
@@ -44,13 +44,25 @@ jobs:
 
   binary_size_metrics:
     needs: check
-    if: always() && github.repository == 'Firebase/firebase-ios-sdk' && (github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.pull_request.merged)
+    if: always() && github.event.pull_request.head.repo.full_name == github.repository && github.event.pull_request.merged
```

Comment:
Could you please add a comment explaining what the check does. If it read it correctly it limits the job to the post-pr of the PR target repo. Does it mean it will be run for forks like FirebasePrivate as well? If yes, will it be a problem?